lib/sda/overlay.go

```diff
@@ -0,0 +1,60 @@
+package sda
+
+import (
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/satori/go.uuid"
+)
+
+/*
+Overlay keeps all trees and entity-lists for a given host. It creates
+Nodes and ProtocolInstances upon request and dispatches the messages.
+*/
+
+type Overlay struct {
+	host        *Host
+	nodes       map[Token]*Node
+	trees       map[uuid.UUID]*Tree
+	entityLists map[uuid.UUID]*EntityList
+}
+
+// NewOverlay creates a new overlay-structure
+func NewOverlay(h *Host) *Overlay {
+	return &Overlay{
+		host:        h,
+		nodes:       make(map[Token]*Node),
+		trees:       make(map[uuid.UUID]*Tree),
+		entityLists: make(map[uuid.UUID]*EntityList),
+	}
+}
+
+// TransmitMsg takes a message received from the host and treats it. It might
+// - ask for the identityList
+// - ask for the Tree
+// - create a new protocolInstance
+// - pass it to a given protocolInstance
+func (o *Overlay) TransmitMsg(msg *SDAData) error {
+	node, ok := o.nodes[*(msg.To)]
```

Comment:
We can directly take msg.To.Id(), so we avoid redirection and it's quicker and more elegant.